### PR TITLE
PLAT-24475 support chapter naming policy for multi clip concat

### PR DESCRIPTION
**Pull request by salma2222**

### CODE REVIEW

File: plugins/cue_points/base/lib/kCuePointManager.php [hunk 1]
```diff
@@ -97,20 +97,43 @@ protected static function handleConcatAfterMultiClipJobFinished($dbBatchJob, $da
...
+			$entry = entryPeer::retrieveByPK($sourceEntryId);
+		}
+		switch ($chapterNamePolicy)
+		{
+			case ChapterNamePolicy::NUMERICAL:
+				$chapterIndex++;
+				return "Chapter $chapterIndex";
+
+			case ChapterNamePolicy::BY_ENTRY_ID:
+				return $entry->getId();
```
Comment (inbal-ben-david) [inline plugins/cue_points/base/lib/kCuePointManager.php:129]: need to validate the entry returned before accessing its property

File: plugins/cue_points/base/lib/kCuePointManager.php [hunk 2]
```diff
@@ -97,20 +97,43 @@ protected static function handleConcatAfterMultiClipJobFinished($dbBatchJob, $da
...
 	}
 
+	protected static function getChapterNameByPolicy($sourceEntryId, $chapterIndex, $chapterNamePolicy)
+	{
+		$entry = null;
+		if($chapterNamePolicy != ChapterNamePolicy::NUMERICAL)
+		{
+			$entry = entryPeer::retrieveByPK($sourceEntryId);
+		}
+		switch ($chapterNamePolicy)
```
Comment (inbal-ben-david) [inline plugins/cue_points/base/lib/kCuePointManager.php:122]: if not set it gets as null and falls to the default case?

Comment (salma2222) [reply plugins/cue_points/base/lib/kCuePointManager.php:122]: yes

File: plugins/cue_points/base/lib/kCuePointManager.php (hunk 1 above)
Comment (salma2222) [reply plugins/cue_points/base/lib/kCuePointManager.php:133]: added

**Review by inbal-ben-david — APPROVED** [review]

**Pull request merged**